### ISSUE TITLE
The 'noun_chunks' syntax iterator is not implemented for language 'en'.

### ISSUE THREAD

**Issue by spattanayak1** [explosion/spaCy#6891]
I have used en_core_web_sm

```
def noun_chunk_identification(raw_tweet):
  global ignore_list
  global spacy
  doc = nlp(raw_tweet)
  append_list = list(doc.noun_chunks)
  append_list = [" ".join([str(y) for y in str(x).split() if len(str(y)) > 1 and str(y).lower() not in ignore_list]) for x in append_list if len(str(x)) > 2 and str(x).lower() not in ignore_list]
  #print (append_list)
  return np.unique(append_list).tolist()  
```

```
noun_chunk_identification_fn = udf(noun_chunk_identification, ArrayType(StringType()))
trend_data2 = trend_data1.withColumn("Noun_Chunks", noun_chunk_identification_fn(col("body")))
display(trend_data2)
```

getting the error:

```
 File "spacy/tokens/doc.pyx", line 833, in noun_chunks
NotImplementedError: [E894] The 'noun_chunks' syntax iterator is not implemented for language 'en'.
```

**Comment by adrianeboyd** [comment]
Hi, hmm, I can't reproduce this. Can you show how you're loading the model and provide more info about your env with `python -m spacy info --markdown`?

**Comment by spattanayak1** [comment]
import spacy
!python -m spacy download en_core_web_sm
nlp = spacy.load("en_core_web_sm")
ignore_list = ["you","myself","they","what","she","he","amp","who","them","her","don","mine"]

def noun_chunk_identification(raw_tweet):
  
  doc = nlp(raw_tweet)
  append_list = list(doc.noun_chunks)
  append_list = [" ".join([str(y) for y in str(x).split() if len(str(y)) > 1 and str(y).lower() not in ignore_list]) for x in append_list if len(str(x)) > 2 and str(x).lower() not in ignore_list]
  
  return append_list

x=spark.createDataFrame(
    [
        (1, 'foo'), # create your data here, be consistent in the types.
        (2, 'bar'),
    ],
    ['id', 'txt'] # add your columns label here
)

from  pyspark.sql.functions import udf, col

from pyspark.sql.types import * # to solve error "name 'ArrayType' is not defined"
import pyspark.sql.functions as F
from pyspark.sql.functions import udf
from datetime import datetime
from pyspark.sql.types import *

noun_chunk_identification_fn = udf(noun_chunk_identification, StringType())
x = x.withColumn("Noun_Chunks", noun_chunk_identification_fn(col("txt")))
display(x)

**Comment by svlandeg** [comment]
This should definitely work on newer versions of spaCy.

Can you run both these commands on the CLI and paste the result?

```
python -m spacy info --markdown
```
```
python -m spacy validate
```

**Comment by spattanayak1** [comment]
I am executing our code in databricks environment.

%sh python -m spacy info --markdown

******************************************************************************
## Info about spaCy

- **spaCy version:** 3.0.1
- **Platform:** Linux-4.15.0-1092-azure-x86_64-with-Ubuntu-18.04-bionic
- **Python version:** 3.7.5
*******************************************************************************

%sh python -m spacy validate
*****************************************************************************
⠙ Loading compatibility table...
✔ Loaded compatibility table

================= Installed pipeline packages (spaCy v3.0.1) =================
ℹ spaCy installation:
/databricks/python3/lib/python3.7/site-packages/spacy

No pipeline packages found in your current environment.
**********************************************************************************************

**Comment by spattanayak1** [comment]
%sh python -m spacy info --markdown

## Info about spaCy

- **spaCy version:** 3.0.1
- **Platform:** Linux-4.15.0-1092-azure-x86_64-with-Ubuntu-18.04-bionic
- **Python version:** 3.7.5
- **Pipelines:** en_core_web_sm (3.0.0)

%sh python -m spacy validate

⠙ Loading compatibility table...
✔ Loaded compatibility table

================= Installed pipeline packages (spaCy v3.0.1) =================
ℹ spaCy installation:
/databricks/python3/lib/python3.7/site-packages/spacy

NAME             SPACY            VERSION                            
en_core_web_sm   >=3.0.0,<3.1.0   3.0.0   ✔



Note: Still we are getting the same issue

**Comment by svlandeg** [comment]
That's weird, it should definitely work on 3.0.1 - and Adriane and myself double checked in our local environments.

Does the databricks installation use virtual environments? I worry that perhaps an older installation of spaCy is messing up the installation, and that at some point the wrong `spacy` is being called into. Could you start from a clean, fresh virtual env?

**Comment by mcPear** [comment]
Hi, I have same error but for 'pl'. Is it really not implemented or it's analogous to spattanayak1's case? I load pl_core_news_lg-3.0.0.

## Info about spaCy

- **spaCy version:** 3.0.0
- **Platform:** Linux-5.9.3-050903-generic-x86_64-with-glibc2.29
- **Python version:** 3.8.5

**Comment by svlandeg** [comment]
@mcPear : the `noun_chunks` is in fact not implemented for `pl`, so that's actually a correct warning you're getting (and not similar to the original problem in the first post) - the functionality is (unfortunately) not there for Polish. But we always appreciate PRs for these kind of language-specific contributions!

**Comment by spattanayak1** [comment]
Dear Team,

This is a clear spacy issue with latest version. 

specifying the version spacy==2.3.5  - This is working fine,
But with latest version it is throwing error.

**Comment by svlandeg** [comment]
Hi @spattanayak1, I definitely believe that you are getting this error, but several of us have double checked on the latest 3.0.1 and are not seeing this error - further more the implementation of the relevant syntax iterator seems to just be there in the source code.

Like I said, this could be due to some corruption in the environment you're running this in. The corruption could be in the newer environment, and not in the old. It's very difficult for us to debug. Have you tried starting from a clean, fresh virtual env?

**Comment by avbelic** [comment]
I'm having the exact same error with spacy==3.0.1 but only when I do parallel processing
So this works for me:
`import spacy
nlp = spacy.load('en_core_web_lg')
doc = nlp("this is a test. That is another test.")
list(doc.noun_chunks)`

but once I use a parallelization routine with pandas dataframe and pathos.multiprocessing to compute Spacy nlp objects I get this error. Same routine worked with spacy==2.x.x. Any thoughts?

**Comment by adrianeboyd** [comment]
Thanks, the info about multiprocessing vs. not was a helpful clue. This is indeed a bug and the problem is that `noun_chunks` isn't being preserved when the doc is pickled. We will get this fixed in the next patch release of v3.0.
